### PR TITLE
[release-1.0]executor: support admin check index

### PR DESCRIPTION
**Pull request by coocood**
List the index values and handle values between multiple int ranges.

### CODE REVIEW

**Comment by coocood** [comment]
@zimulala @winkyao PTAL

File: plan/plans.go [hunk 1]
```diff
@@ -46,6 +46,16 @@ type CheckTable struct {
 	Tables []*ast.TableName
 }
 
+// CheckIndex is used for checking index data, output the index values that handle within begin and end.
+type CheckIndex struct {
+	basePlan
+
+	Table     *ast.TableName
+	IndexName string
+	Begin     int64
```
Comment (shenli) [inline plan/plans.go]: So we could only check index with integer column?

File: executor/admin.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	init    bool
+
+	result   distsql.NewSelectResult
+	partial  distsql.NewPartialResult
+	typesMap map[int64]*types.FieldType
+	cols     []*model.ColumnInfo
+}
+
+// Next implements the Executor Next interface.
+func (e *CheckIndexExec) Next() (Row, error) {
```
Comment (shenli) [inline executor/admin.go]: Do we need to implement NextChunk() interface?

Comment (coocood) [reply executor/admin.go]: release-1.0 doesn't has `NextChunk` interface.

File: plan/plans.go (hunk 1 above)
Comment (coocood) [reply plan/plans.go]: This is the handle range, not the index value range.

File: executor/admin.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	return rowData, errors.Trace(err)
+}
+
+func (e *CheckIndexExec) getNextInRangeIndex() ([]types.Datum, error) {
+	for {
+		if e.partial == nil {
+			var err error
+			e.partial, err = e.result.Next()
+			if err != nil {
+				return nil, nil
```
Comment (zimulala) [inline executor/admin.go]: Why we don't return the `err`?

File: executor/admin.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+	return dagReq, nil
+}
+
+func (e *CheckIndexExec) constructIndexScanPB() *tipb.Executor {
+	tableColumns := e.table.Meta().Columns
+	var columns []*model.ColumnInfo
+	for _, col := range e.index.Meta().Columns {
+		columns = append(columns, tableColumns[col.Offset])
+	}
+	columns = append(columns, &model.ColumnInfo{
```
[the remark this answers is not in the document]
Comment (coocood) [reply executor/admin.go]: the function `ColumnsToProto` only accepts column info args.

**Comment by coocood** [comment]
/run-all-tests tikv=release-1.0 tidb-test=release-1.0 tidb-private-test=release-1.0

**Comment by coocood** [comment]
/run-integration-common-test tidb-test=release-1.0 tidb-private-test=release-1.0

**Comment by coocood** [comment]
/run-integration-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 pd=release-1.0

**Comment by coocood** [comment]
/run-integration-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 pd=release-1.0 tikv=release-1.0

**Comment by coocood** [comment]
/run-integration-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 pd=release-1.0 tikv=release-1.0

**Comment by coocood** [comment]
/run-integration-common-test tikv=release-1.0 tidb-test=release-1.0 tidb-private-test=release-1.0

**Comment by coocood** [comment]
/run-integration-common-test tikv=release-1.0 tidb-test=release-1.0 tidb-private-test=release-1.0

**Comment by shenli** [comment]
Any update?

**Comment by zhexuany** [comment]
/run-integration-common-test tikv=release-1.0 tidb-test=release-1.0 tidb-private-test=release-1.0 

@ciscoxll Please fix ci if it is still failing.

File: executor/builder.go [hunk 5]
```diff
@@ -214,6 +217,33 @@ func (b *executorBuilder) buildCheckTable(v *plan.CheckTable) Executor {
...
+		baseExecutor: newBaseExecutor(v.Schema(), b.ctx),
+		handleRanges: v.HandleRanges,
+		table:        tb,
+		is:           b.is,
+	}
+	idxName := strings.ToLower(v.IndexName)
+	for _, idx := range tb.Indices() {
+		if idx.Meta().Name.L == idxName {
+			e.index = idx
+			e.startKey = make([]types.Datum, len(e.index.Meta().Columns))
```
[the remark this answers is not in the document]
Comment (zhexuany) [reply executor/builder.go]: @ciscoxll  Please address reviewer's comment.

**Comment by ciscoxll** [comment]
/run-integration-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 pd=release-1.0 tikv=release-1.0

**Comment by ciscoxll** [comment]
/run-integration-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 pd=release-1.0 tikv=release-1.0

**Comment by coocood** [comment]
/run-integration-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 pd=release-1.0 tikv=release-1.0

**Pull request closed without merging**